### PR TITLE
fix(autocomplete): support ng-required

### LINKED ISSUE #2949: mdAutocomplete should support ng-required
mdAutocomplete does not pass ng-required to the underlying input. This causes problems performing form validation.

### PR DESCRIPTION
**Pull request by loic**
Closes #2949

### CODE REVIEW

**Comment by topherfangio** [comment]
Can you please provide a test case that shows exactly what this fixes?

**Comment by loic** [comment]
Added tests, unless I missed it, required wasn't tested for. I couldn't find any tests for the rest of the validation stack either, e.g. minlength, maxlength, ng-messages, ng-disabled, etc.

**Comment by EladBezalel** [comment]
@loic please rebase

**Comment by loic** [comment]
It looks like 32ba92f44e5ffcb18f0663074d7c9c1913e72836 addressed the issue. I rebased the tests in case they are of interest, otherwise feel free to close the PR.

**Comment by DevVersion** [comment]
@loic Thanks for the updates. I like those tests, they're additionally testing the behavior of required autocomplete's  inside of the form validation.

File: src/components/autocomplete/autocomplete.spec.js [hunk 1]
```diff
@@ -1154,4 +1154,66 @@ describe('<md-autocomplete>', function() {
     }));
   });
 
+  describe('required', function () {
```
Comment (DevVersion) [inline src/components/autocomplete/autocomplete.spec.js]: Please notice, that we already have a section for `required`.

https://github.com/angular/material/blob/master/src/components/autocomplete/autocomplete.spec.js#L1035

Comment (loic) [inline src/components/autocomplete/autocomplete.spec.js]: Good catch, it didn't exist when I made the original PR, let me move it around.

Comment (DevVersion) [inline src/components/autocomplete/autocomplete.spec.js]: Understood, thanks :)

**Comment by loic** [comment]
Actually both seem already tested https://github.com/angular/material/blob/master/src/components/autocomplete/autocomplete.spec.js#L1066-L1112. I think we can close this PR.

**Comment by DevVersion** [comment]
@loic I agree, thanks for your contribution. I'm sorry, we went ahead with another PR.

**Pull request closed without merging**